src/analyzer/protocol/ssl/SSL.cc

```diff
@@ -98,4 +174,166 @@ void SSL_Analyzer::Undelivered(uint64_t seq, int len, bool orig)
 	interp->NewGap(orig, len);
 	}
 
+void SSL_Analyzer::SetSecret(const u_char* data, int len)
+	{
+	// FIXME: Is this the proper way to initialize a zeek::StringVal?
+	//        the alternative requires a cast: new zeek::StringVal(len, data)
+	secret = new zeek::StringVal(new zeek::String(data, len, true));
+	}
+
+void SSL_Analyzer::SetKeys(const u_char* data, int len)
+	{
+	keys = new zeek::StringVal(new zeek::String(data, len, true));
+	}
+
+bool SSL_Analyzer::TryDecryptApplicationData(int len, const u_char* data, bool is_orig, uint8_t content_type, uint16_t raw_tls_version)
+	{
+	// Unsupported cipher suite. Currently supported:
+	// - TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 == 0xC030
+	auto cipher = handshake_interp->chosen_cipher();
+	if ( cipher != 0xC030 )
+		{
+		//printf("Unsupported cipher suite: %d\n", cipher);
+		return false;
+		}
+
+	// Neither secret or key present: abort
+	if ( secret->Len() == 0 && keys->Len() == 0 )
+		{
+		// FIXME: this is just for debugging
```

Comment:
those printfs should obviously go. It would be neat to instead make them into debug messages, so e.g....

```
DBG_LOG(DBG_ANALYZER, "Could not decrypt packet...")
```